### PR TITLE
Add test for divider attributes

### PR DESCRIPTION
**Pull request by devkant**
Fixes: #194

### CODE REVIEW

**Comment by sumanthvrao** [comment]
@devkant Thanks for the PR! One quick thing I would like to point out is that we don't make use of `git merge` workflow. We use a rebase based work-flow (This avoids the merge commits like the one occurring here). More details can be found here - https://zulip.readthedocs.io/en/latest/git/pull-requests.html#id1. You could `squash` the first two commits as well.

**Comment by devkant** [comment]
@neiljp  It would be great if could review it.

**Comment by devkant** [comment]
@neiljp I have used side_effect and objects of a temporary class button to create a list of buttons. It fails the test if we remove the attributes from divider. Please review it.

File: tests/ui/test_ui_tools.py [hunk 1]
```diff
@@ -1001,6 +1001,42 @@ def test_topics_view(self, mocker, stream_button, width=40):
...
+
+        def altdivider(char):
+            obj = button()
+            return obj
+
+        def altlinebox(view, *args, **kwargs):
+            return view
+
+        def altstreamsview(btnlist, *args, **kwargs):
+            return btnlist
```
Comment (neiljp) [inline tests/ui/test_ui_tools.py]: You shouldn't need these classes/functions, as that's either what the mock does for you, or you actually want to test the code itself.

File: tests/ui/test_ui_tools.py [hunk 2]
```diff
@@ -1001,6 +1001,42 @@ def test_topics_view(self, mocker, stream_button, width=40):
...
+
+        stream_button = mocker.patch(VIEWS + '.StreamButton',
+                                     side_effect=altbutton)
+        stream_view = mocker.patch(VIEWS + '.StreamsView',
+                                   side_effect=altstreamsview)
+        divider = mocker.patch(VIEWS + '.urwid.Divider',
+                               side_effect=altdivider)
```
Comment (neiljp) [inline tests/ui/test_ui_tools.py]: The above 3 mocks shouldn't be necessary, as we're wanting to test the actual code, not a mock. Also, for future reference, you don't test/use the mocks later, so you don't need to assign them to variables.

File: tests/ui/test_ui_tools.py [hunk 3]
```diff
@@ -1001,6 +1001,42 @@ def test_topics_view(self, mocker, stream_button, width=40):
...
+        line_box = mocker.patch(VIEWS + '.urwid.LineBox')
+        mocker.patch(STREAMBUTTON + ".mark_muted")
+        left_col_view = LeftColumnView(width, self.view)
+        for i in left_col_view.view.stream_w:
+            assert hasattr(i, 'stream_name') and hasattr(i, 'stream_id')
```
Comment (neiljp) [inline tests/ui/test_ui_tools.py]: While pytest is pretty clever at identifying which element of an expression is failing, it's probably clearer if you have these on separate lines.

File: tests/ui/test_ui_tools.py [hunk 4]
```diff
@@ -1001,6 +1001,42 @@ def test_topics_view(self, mocker, stream_button, width=40):
...
+        stream_button = mocker.patch(VIEWS + '.StreamButton',
+                                     side_effect=altbutton)
+        stream_view = mocker.patch(VIEWS + '.StreamsView',
+                                   side_effect=altstreamsview)
+        divider = mocker.patch(VIEWS + '.urwid.Divider',
+                               side_effect=altdivider)
+        line_box = mocker.patch(VIEWS + '.urwid.LineBox')
+        mocker.patch(STREAMBUTTON + ".mark_muted")
+        left_col_view = LeftColumnView(width, self.view)
+        for i in left_col_view.view.stream_w:
```
Comment (neiljp) [inline tests/ui/test_ui_tools.py]: Have you looked at the `stream_w`? You should be able to `print` it; `pytest` should display it in the details at the end, and/or you can use the `-s` option to show the output during each individual 'dot' of the test. I'm not convinced it contains what you want.

**Comment by neiljp** [comment]
@devkant Thanks for giving this a go :+1:

**Pull request closed without merging**